src/org/labkey/test/components/glassLibrary/components/BaseReactSelect.java

```diff
@@ -90,6 +90,18 @@ public boolean isLoading()
         return Locators.loadingSpinner.findElementOrNull(getComponentElement()) != null;
     }
 
+    public boolean isLoaded()
+    {
+        return !isLoading() && !getOptions().contains("Loading...");
+    }
+
+    public T waitForLoaded(int waitMsec)
+    {
+        _wrapper.waitFor(()-> isLoaded(),
+                "Took too long for to become loaded", waitMsec);
+        return (T) this;
+    }
```

Comment:
I don't see a reason for any of the loaded/loading methods to be public. Any interaction methods should handle waiting internally.
```suggestion
    protected void waitForLoaded(int waitMsec)
    {
        _wrapper.waitFor(()-> isLoaded(),
                "Took too long for select to load: " + getComponentElement().toString(), waitMsec);
    }
```

Reply:
Okay.  I'll consolidate logic into a waitForLoaded() method and call it from interaction/entry points
If the logic is different in derived classes we can override there